src/main/java/org/sonarlint/intellij/core/SonarLintEngineFactory.java

```diff
@@ -151,6 +158,19 @@ private static URL findEmbeddedCFamilyPlugin(Path pluginsDir) {
     }
   }
 
+  @CheckForNull
+  private static URL findEmbeddedCsharpPlugin(Path pluginsDir) {
+    try {
+      List<URL> pluginsUrls = findFilesInDir(pluginsDir, "sonarlint-omnisharp-plugin-*.jar", "Found CSharp plugin: ");
+      if (pluginsUrls.size() > 1) {
+        throw new IllegalStateException("Multiple plugins found");
+      }
+      return pluginsUrls.size() == 1 ? pluginsUrls.get(0) : null;
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
```

Comment:
I think this method is pretty similar to findEmbeddedCFamilyPlugin, you might want to factorize